### PR TITLE
feat: config Translate

### PR DESCRIPTION
**Pull request by Lanng21**
# Description

- Confìg Translate 
- Update columns on Table
## Detailed PR Verification Procedure(s)

- None

## Ticket

- [GS-47](https://tito-training.atlassian.net/browse/GS-47)

## Application Checklist

- [x] **JIRA Linked**: Author has followed template and has a title in the following format:
  > GS-47: Generic summary of work done in PR
- [ ] **No Breaking Changes**: Author has confirmed all API contract changes are non-breaking.
- [ ] **Changes Unit Tested**: Author has unit tested new code to prove it works.
- [ ] **Application Passes All Tests**: Author ensures this pull-request passes `npm test`.
- [x] **Author Self Reviewed**: Author of pull-request has self-reviewed this code.
- [x] **Author is Assignee**: Author has assigned self as assignee of pull request.

## Security Checklist

- [ ] **Code Review for Security:** Developer has self-reviewed this code, based on OWASP Top-10 vulnerabilities and general environment security attributes, to avoid dversely impact the security of the system/environment/application, to the best of their abilities.
- [ ] **Backout Procedures:** Developer has considered how the changes presented in this pull request can be replaced with a prior version of these changes, fixed going forward, and/or implemented to be backwards compatible.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -83,5 +87,9 @@
     "src/**/*.{ts,tsx}": [
       "eslint --fix --max-warnings=0"
     ]
-  }
+  },
+  "main": "index.js",
```
Comment (baonguyen26) [inline package.json]: remove

File: src/views/customer/components/AddCustomer/index.tsx [hunk 2]
```diff
@@ -16,24 +17,13 @@ import { CaretUpOutlined, CaretDownOutlined } from '@ant-design/icons'
...
-  demand,
-  email,
-  employeeSupport,
-  fullName,
-  gender,
-  note,
-  OPTIONSELECT,
-  phoneNumber,
-} from '@/constants/Customer'
+import { OPTIONSELECT } from '@/constants/Customer'
```
Comment (baonguyen26) [inline src/views/customer/components/AddCustomer/index.tsx]: naming confused

File: src/views/customer/components/AddCustomer/index.tsx [hunk 3]
```diff
@@ -103,48 +99,60 @@ const AddCustomer: React.FC<AddCustomerProps> = ({ onClose }) => {
         <hr />
         <Row gutter={16}>
           <Col span={12}>
-            <Form.Item name='Date' label={birthday}>
-              <DatePicker suffixIcon={<Calendar />} placeholder={birthday} />
+            <Form.Item name='Date' label={t('addCustomers.birthday')}>
+              <DatePicker
+                suffixIcon={<Calendar />}
+                format='DD/MM/YYYY'
```
Comment (baonguyen26) [inline src/views/customer/components/AddCustomer/index.tsx]: move `DD/MM/YY` to constant, it should common

File: src/views/customer/components/AddCustomer/index.tsx [hunk 4]
```diff
@@ -103,48 +99,60 @@ const AddCustomer: React.FC<AddCustomerProps> = ({ onClose }) => {
...
+              <Input placeholder={t('addCustomers.address.placeholder')} />
             </Form.Item>
           </Col>
         </Row>
         <Row gutter={16}>
           <Col span={12}>
             <Form.Item
               name='phone_number'
-              label={phoneNumber.label}
+              label={t('addCustomers.phoneNumber.label')}
```
Comment (baonguyen26) [inline src/views/customer/components/AddCustomer/index.tsx]: `addCustomers.phoneNumber.label` is too long

File: src/views/customer/components/AddCustomer/index.tsx [hunk 5]
```diff
@@ -16,24 +17,13 @@ import { CaretUpOutlined, CaretDownOutlined } from '@ant-design/icons'
...
-  OPTIONSELECT,
-  phoneNumber,
-} from '@/constants/Customer'
+import { OPTIONSELECT } from '@/constants/Customer'
 
 interface AddCustomerProps {
   onClose: () => void
 }
 const AddCustomer: React.FC<AddCustomerProps> = ({ onClose }) => {
+  const { t } = useTranslation('customer')
```
Comment (baonguyen26) [inline src/views/customer/components/AddCustomer/index.tsx]: `const { t } = useTranslation('customer addCustomers')`
=> Ex: label={t('fullName.label')}

File: src/views/customer/components/Empty/index.tsx [hunk 6]
```diff
@@ -2,14 +2,15 @@ import { memo, useCallback } from 'react'
...
 import Button from '@/components/common/Button'
 import Empty from '@/components/common/Empty'
-//constants
-import { Customer } from '@/constants/Customer'
+
 //icon
 import emptyImg from '@/assets/icons/Client.svg'
 import { ReactComponent as AddCustomer } from '@/assets/icons/AddCustomer.svg'
 import { ReactComponent as FileImport } from '@/assets/icons/FileImport.svg'
+import { useTranslation } from 'react-i18next'
```
Comment (baonguyen26) [inline src/views/customer/components/Empty/index.tsx]: order import, it should import before assets

File: yarn.lock [hunk 7]
```diff
@@ -1097,7 +1097,7 @@
   resolved "https://registry.yarnpkg.com/@babel/regjsgen/-/regjsgen-0.8.0.tgz#f0ba69b075e1f05fb2825b7fad991e7adbb18310"
```
Comment (baonguyen26) [inline yarn.lock]: move this file to gitignore

File: src/views/customer/components/AddCustomer/index.tsx [hunk 8]
```diff
@@ -1,10 +1,11 @@
 import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
 import TextArea from 'antd/lib/input/TextArea'
 import { Col, DatePicker, Form, Radio, Row, Select } from 'antd'
 import { Gender } from '@/types/gender'
+import { ICustomer } from '@/types/Customer'
```
Comment (baonguyen26) [inline src/views/customer/components/AddCustomer/index.tsx:6]: break line

File: src/views/customer/locales/en.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      note: {
+        label: 'Note',
+        placeholder: 'Enter note',
+      },
+      btn: {
+        cancel: 'Cancel',
+        submit: 'Add Customer',
+      },
+    },
+    table: {
```
Comment (baonguyen26) [inline src/views/customer/locales/en.ts]: It should move to constant for table, we can reuse it

**Review by duong56 — COMMENTED** [review]
Other than that comment look good to me

**Review by duong56 — APPROVED** [review]
Other than that comment look good to me

**Comment by duong56** [comment]
please check comment

File: src/views/customer/locales/en.ts (hunk 9 above)
Comment (Lanng21) [reply src/views/customer/locales/en.ts]: But this table is only used in customer and Each table has different fields. i think so

File: src/constants/Customer.ts [hunk 10]
```diff
@@ -1,68 +1,3 @@
...
-  address,
-  gender,
-  birthday,
-  fullName,
-  OPTIONSELECT,
-  Table,
-}
+const Demand = 'test'
+const formatDate = 'DD/MM/YYYY'
+export { Demand, formatDate }
```
Comment (baonguyen26) [inline src/constants/Customer.ts]: `export default cutomer`

Comment (baonguyen26) [reply src/constants/Customer.ts]: import { Demand, formatDate } from 'customer'

**Review by baonguyen26 — APPROVED** [review]

**Pull request merged**